### DISCUSSION TITLE
Finding if a car is sleeping without waking it

### DISCUSSION THREAD

**Discussion by SylvainGa** [timdorr/tesla-api#525] [Q&A]
Hi, Is there a way to see if a car is sleeping without waking it?

I'm thinking of adding a feature to the Garmin watch face I'm using to display my car's current charge level but to prevent the car from being kept awake, only poll when it's currenty awake.

Thanks.

**Comment by PCRover** [comment]
I send a status request and look at the result. The vehicle should not wake
until you send it a wake up request.

**Reply by SylvainGa** [reply]
Thanks for the quick reply, why API call is a 'status request' ?

**Reply by SylvainGa** [reply]
Is it `GET /api/1/vehicles/{id}/data_request/vehicle_state` ?

**Comment by davidhodge** [comment]
This should work. The harder trick is checking if a car is actively doing something when it's online without that keeping it online for longer!

**Reply by SylvainGa** [reply]
I understand. So stop polling when charging_state is no longer active should do the trick for this if that's all I'm interested in then?

**Reply by SylvainGa** [reply]
Hmm, here's what I found:

`GET /api/1/vehicles/{id}/data_request/charge_state` does keep the vehicle awake, but `GET /api/1/vehicles/{id}/vehicle_data` do NOT. I have a Garmin watch and modified the [Crystal watch face](https://github.com/SylvainGa/crystal-face) to include my car's battery level to the main screen. I was first using the ...vehicle_data call and the car was falling asleep, even at a polling interval of 5 minutes. Since this call was taking 4 times the memory of a ...data_request/charge_state call (and the watch memory is quite limited), I switched to that one. 

Bad move, the car no longer falls asleep. Tried for several cycles and it stayed awake. I switched it back to vehicle_data and it fell asleep. I haven't tried the other data_request/* calls but charging_state (the one I was interesting in :-( ) do keep it awake.

**Reply by SylvainGa** [reply]
And here's proof that vehicle_data do NOT keep the vehicle awake. It was able to fall asleep, even while being queried every five minutes and stayed asleep.
```

15:21:02 : onReceiveVehicleData responseCode is 200
15:26:01 : onReceiveVehicleData responseCode is 200
15:30:00 : onReceiveVehicleData responseCode is 408
15:36:00 : onReceiveVehicleData responseCode is 408
15:41:00 : onReceiveVehicleData responseCode is 408
```

**Comment by PCRover** [comment]
Pretty much any request will respond with a "vehicle not available" or
other fail response if the vehicle is asleep.

**Reply by SylvainGa** [reply]
Thanks
